components/src/dynamo/thunderagent_router/capacity.py

```
@@ -57,25 +69,92 @@ def snapshot(self) -> dict[int, int]:
+
     def live_worker_ids(self) -> set[int]:
-        """Return workers currently registered for the generate endpoint."""
+        """Return workers currently registered for the generate endpoint.
+
+        Deliberately worker-granular, not replica-granular: liveness is a property of
+        the instance (``Client.instance_ids()`` reports nothing finer), and a stale-pin
+        check only needs to know whether the process is gone.
+        """
         try:
             return set(self._client.instance_ids())
         except Exception as exc:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)